### PR TITLE
Fix assumptions about tmpfs in metrics du tests

### LINKED ISSUE #18576: Metrics du tests broken on F22
Broken tests for me on Fedora 22:

```
 hack/test-go.sh pkg/volume{,/host_path,/empty_dir}                                                                                                                                                               
Running tests for APIVersion: v1,extensions/v1beta1,metrics/v1alpha1 with etcdPrefix: registry
+++ [1211 10:40:39] Running tests without code coverage
--- FAIL: TestMetricsDuGetCapacity (0.00s)
        metrics_du_test.go:41: Expected Used 0 for empty directory to be 4096.
        metrics_du_test.go:60: Unexpected Used for directory with file.  Expected 8192, was 4096.
FAIL
FAIL    k8s.io/kubernetes/pkg/volume    0.011s
--- FAIL: TestMetrics (0.00s)
        host_path_test.go:309: Expected Used &{0 %!d(resource.Format=BinarySI)} to be 4096
FAIL
FAIL    k8s.io/kubernetes/pkg/volume/host_path  0.009s
--- FAIL: TestMetrics (0.00s)
        empty_dir_test.go:307: Expected Used 0 to be 4096
FAIL
FAIL    k8s.io/kubernetes/pkg/volume/empty_dir  0.017s
!!! Error in hack/test-go.sh:240
  'return ${rc}' exited with status 1
Call stack:
  1: hack/test-go.sh:240 main(...)
Exiting with status 1
```

### PR DESCRIPTION
**Pull request by pmorie**
Fixes #18576 

@thockin @vishh @pwittrock @rootfs @swagiaal

### CODE REVIEW

**Comment by pmorie** [comment]
@k8s-bot test this

File: pkg/volume/testing.go [hunk 1]
```diff
@@ -227,3 +231,22 @@ func (fd *FakeDeleter) Delete() error {
 func (fd *FakeDeleter) GetPath() string {
 	return fd.path
 }
+
+// FindEmptyDirectoryUsageOnTmpfs finds the expected usage of an empty directory existing on
+// a tmpfs filesystem on this system.
+func FindEmptyDirectoryUsageOnTmpfs() (*resource.Quantity, error) {
+	tmpDir, err := ioutil.TempDir(os.TempDir(), "metrics_du_test")
```
Comment (eparis) [inline pkg/volume/testing.go]: what makes you think this is on tmpfs?

File: pkg/volume/metrics_du_test.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ import (
 	"testing"
 )
 
+const expectedBlockSize = 4096
```
Comment (eparis) [inline pkg/volume/metrics_du_test.go]: while VERY usually correct, not all blocks are 4k.   should the test get the block size from stat info and just accept either 0 or 1 blocks in size?  maybe that doesn't help either...

Comment (rootfs) [inline pkg/volume/metrics_du_test.go]: See if my [comment](https://github.com/kubernetes/kubernetes/pull/18232/files#r47375263) can help.

**Comment by pmorie** [comment]
@k8s-bot test this

File: pkg/volume/testing.go (hunk 1 above)
Comment (pmorie) [inline pkg/volume/testing.go]: `os.TempDir()` gives you /tmp on linux...

Comment (pmorie) [inline pkg/volume/testing.go]: I suppose you're right though, it's really expected usage of an empty temporary directory however that might be defined on your system.

File: pkg/volume/testing.go [hunk 3]
```diff
@@ -227,3 +231,22 @@ func (fd *FakeDeleter) Delete() error {
...
+	if err != nil {
+		return nil, err
+	}
+	out, err := exec.Command("nice", "-n", "19", "du", "-s", "-B", "1", tmpDir).CombinedOutput()
```
Comment (vishh) [inline pkg/volume/testing.go]: Can we reuse existing `du` code? And why is the usage not `0` to begin with?

File: pkg/volume/metrics_du_test.go (hunk 2 above)
Comment (pmorie) [inline pkg/volume/metrics_du_test.go]: Thanks, @rootfs, I'll work this in.

**Comment by pmorie** [comment]
@eparis @rootfs If I fix the name of the method that finds the expected usage, is that good enough for now, and we can do the proper computation in a follow up?

File: pkg/volume/testing.go (hunk 3 above)
Comment (pmorie) [inline pkg/volume/testing.go]: @vissh under the debian flavor in GCE, directories in /tmp report 4096 under du. :(

**Comment by pmorie** [comment]
@k8s-bot test this

File: pkg/volume/testing.go (hunk 3 above)
Comment (vishh) [inline pkg/volume/testing.go]: I don't think `/tmp` is a tmpfs on GCE. But it does report 4096 as usage.

File: pkg/volume/testing.go (hunk 1 above)
Comment (vishh) [inline pkg/volume/testing.go]: So this method is not working around a tmpfs issue, but rather a GCE ext4 issue. Should we rename the method appropriately?

**Comment by pmorie** [comment]
I'd really like to get this in and will fix the block size checking with stat once this is in master in a follow-up PR.

File: pkg/volume/testing.go (hunk 1 above)
Comment (pmorie) [inline pkg/volume/testing.go]: Yeah, I'll rename the method to be what it represents.  The issue isn't with any filesystem, really, it's with the assumptions baked into the test.

File: pkg/volume/testing.go (hunk 3 above)
Comment (pmorie) [inline pkg/volume/testing.go]: @vishh you're right:

```
pmorie@du-test:~$ mount
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,relatime,size=10240k,nr_inodes=472940,mode=755)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,relatime,size=760208k,mode=755)
/dev/sda1 on / type ext4 (rw,relatime,data=ordered)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/lib/systemd/s
ystemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=23,pgrp=1,timeout=300,minproto=5,maxproto=5,di
rect)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
mqueue on /dev/mqueue type mqueue (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime)
```

**Comment by pmorie** [comment]
@k8s-bot unit test this

**Comment by pmorie** [comment]
Ugh, I borked up the rebase for this onto master somehow.  Should be sorted now.

**Comment by pmorie** [comment]
@k8s-bot unit test this

**Comment by pmorie** [comment]
@k8s-oncall Could the on-call dev take a look at this and consider for merge?  This has a lot of us at Red Hat busted.

**Comment by pmorie** [comment]
@saad-ali thanks for merge bro

**Comment by saad-ali** [comment]
@pmorie Anytime :)

**Pull request merged**